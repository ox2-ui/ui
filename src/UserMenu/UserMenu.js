import PropTypes from 'prop-types';
import React from 'react';
import Button from '@ox2/button/Button';

/**
 * UserMenu Component
 */
const UserMenu = ({ onDismiss }) =>
  <div className="padding-b:40">
    <Button onClick={() => onDismiss()} btn="raised action">
      Done
    </Button>
    <div>
      USER Lorem ipsum dolor sit amet, consectetur
      adipisicing elit. Itaque, sequi? Consequuntur
      inventore nulla ratione vitae obcaecati voluptatem
      animi molestiae ipsam et adipisci illo reiciendis
      saepe doloremque, quae perspiciatis iure vero.
    </div>
  </div>;

UserMenu.propTypes = {
  onDismiss: PropTypes.func.isRequired,
};

export default UserMenu;
