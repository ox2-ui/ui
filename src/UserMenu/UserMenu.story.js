import React from 'react';
import { storiesOf } from '@storybook/react';
import UserMenu from './UserMenu';

const handler = () => console.log('🍇'); // eslint-disable-line no-console

storiesOf('UserMenu', module).add('default', () =>
  <UserMenu onDismiss={handler} />,
);
