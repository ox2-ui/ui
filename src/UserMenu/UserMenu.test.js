/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import UserMenu from './UserMenu';

const testHandler = () => true;

it('renders correctly', () => {
  const wrapper = mount(
    <UserMenu onDismiss={testHandler} />,
  );

  expect(wrapper).toMatchSnapshot();
});
