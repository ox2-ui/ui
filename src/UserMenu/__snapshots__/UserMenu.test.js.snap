// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<UserMenu
  onDismiss={[Function]}
>
  <div
    className="padding-b:40"
  >
    <Button
      btn="raised action"
      className=""
      onClick={[Function]}
      style={Object {}}
    >
      <button
        className=""
        data-btn="raised action"
        onClick={[Function]}
        style={Object {}}
      >
        Done
      </button>
    </Button>
    <div>
      USER Lorem ipsum dolor sit amet, consectetur adipisicing elit. Itaque, sequi? Consequuntur inventore nulla ratione vitae obcaecati voluptatem animi molestiae ipsam et adipisci illo reiciendis saepe doloremque, quae perspiciatis iure vero.
    </div>
  </div>
</UserMenu>
`;
